### PR TITLE
fix(dev-infra): use commit message validation from @angular/dev-infra-private

### PR DESCRIPTION
**Pull request by josephperrott**

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -153,6 +153,10 @@
     universal-analytics "^0.4.20"
     uuid "^3.3.2"
 
+"@angular/dev-infra-private@angular/dev-infra-private-builds#204e328":
```
Comment (filipesilva) [inline yarn.lock]: Is this right? I don't think the yarn lock should have the package here if it's not on package.json.

Comment (gkalpak) [reply yarn.lock]: It _is_ in [package.json#L49][1] 😕 

[1]: https://github.com/angular/angular/blob/c8c6386a45dd936189126cb82dc00229341b934d/package.json#L49

Comment (filipesilva) [reply yarn.lock]: Oh jeez I missed it, thanks George!

**Review by filipesilva — APPROVED** [review]

**Comment by josephperrott** [comment]
Blocked by https://github.com/angular/angular/pull/36326

**Comment by josephperrott** [comment]
@IgorMinar @gkalpak @filipesilva This is ready for another look as I had to do a few updates to build it from local.

File: dev-infra/commit-message/cli.ts [hunk 2]
```diff
@@ -7,13 +7,27 @@
...
+      .strict()
+      .command(
+          'pre-commit-validate', 'Validate the most recent commit message', {},
+          () => {
+            validateFile('.git/COMMIT_EDITMSG');
+          })
+      .command(
+          'validate-range', 'Validate a range of commit messages', {
+            'range': {
+              description: 'The range of commits check',
```
Comment (IgorMinar) [inline dev-infra/commit-message/cli.ts]: maybe provide an example as well so that it's easier to use?

File: dev-infra/commit-message/cli.ts [hunk 3]
```diff
@@ -7,13 +7,27 @@
  */
 import * as yargs from 'yargs';
 import {validateFile} from './validate-file';
-
```
Comment (gkalpak) [inline dev-infra/commit-message/cli.ts:10]: 😞

File: dev-infra/commit-message/cli.ts (hunk 2 above)
Comment (gkalpak) [inline dev-infra/commit-message/cli.ts:23]:
```suggestion
              description: 'The range of commits to check',
```

File: dev-infra/commit-message/validate-range.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+ * Copyright Google Inc. All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+import {exec} from 'shelljs';
+import {parseCommitMessage, validateCommitMessage, ValidateCommitMessageOptions} from './validate';
+
+// Whether the provided commit is a fixup commit.
+const isNonFixup = (m: string) => parseCommitMessage(m).isFixup;
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range.ts:12]:
```suggestion
const isNonFixup = (m: string) => !parseCommitMessage(m).isFixup;
```

:grin:

File: dev-infra/commit-message/validate-range.ts [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+  const gitLogFormat = `%s%n%n%b${randomValueSeparator}`;
+
+  // Retrieve the commits in the provided range.
+  const result = exec(`git log --reverse --format=${gitLogFormat} ${range}`, {silent: true});
+  if (result.code) {
+    throw new Error(`Failed to get all commits in the range: \n  ${result.stderr}`);
+  }
+
+  // Separate the commits from a single string into individual commits
+  const commits = result.split(randomValueSeparator).map(l => l.trim()).filter(line => line != '');
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range.ts:29]: Nit:

```suggestion
  const commits = result.split(randomValueSeparator).map(l => l.trim()).filter(line => line !== '');
```

File: dev-infra/commit-message/validate-range.ts [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+
+  console.info(`Examining ${commits.length} commit(s) in the provided range: ${range}`);
+
+  // Check if each commit in the range is valid, providing the possible commits that could be
+  // fixed up into.
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range.ts:34]: Into?

File: dev-infra/commit-message/validate-range.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+  const allCommitsInRangeValid = commits.every((m, i) => {
+    const options: ValidateCommitMessageOptions = {
+      disallowSquash: true,
+      nonFixupCommitHeaders: isNonFixup(m) ? undefined : commits.slice(0, i).filter(isNonFixup),
+      ignoreHeaderLength: true,
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range.ts:39]: Doesn't this mean that we can end up with too long commits (if people decide to trick our validation)?
Not the end of the world, but why not keep the previous behavior: Validate on CI for PR-builds only and do not ignore header length?

**Review by gkalpak — CHANGES_REQUESTED** [review]
Yay for deleting code 🤤 🎉

**Review by IgorMinar — APPROVED** [review]
ironically - can you please update the commit message for this change to answer the "why this change?" question?

thanks


lgtm otherwise.

File: dev-infra/commit-message/validate-range.ts [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+    throw new Error(`Failed to get all commits in the range: \n  ${result.stderr}`);
+  }
+
+  // Separate the commits from a single string into individual commits
+  const commits = result.split(randomValueSeparator).map(l => l.trim()).filter(line => !!line);
+
+  console.info(`Examining ${commits.length} commit(s) in the provided range: ${range}`);
+
+  // Check if each commit in the range is valid, providing the possible commits that could be
+  // FIXUPs could be linked to.
```
Comment (gkalpak) [inline dev-infra/commit-message/validate-range.ts:34]: Something is off with this last clause 😕

**Review by gkalpak — APPROVED** [review]
:+1:

**Pull request closed without merging**